kong/runloop/handler.lua

```diff
@@ -59,16 +62,37 @@ end
 
 do
   -- Given a protocol, return the subsystem that handles it
-  local protocol_subsystem = {
-    http = "http",
-    https = "http",
-    tcp = "stream",
-    tls = "stream",
-  }
-
   local router
   local router_version
 
+  local function should_process_route(route)
+    for _, protocol in ipairs(route.protocols) do
+      if constants.PROTOCOLS_WITH_SUBSYSTEM[protocol] == subsystem then
+        return true
+      end
+    end
+  end
+
+  local function get_service_for_route(db, route)
+    local service_pk = route.service
+    if not service_pk then
+      return nil, nil, subsystem
+    end
+
+    local service, err = db.services:select(service_pk)
+    if not service then
+      return nil, "could not find service for route (" .. route.id .. "): " ..
+                  err
+    end
+
+    -- TODO: this should not be needed as the schema should check it already
+    local service_subsystem = constants.PROTOCOLS_WITH_SUBSYSTEM[service.protocol]
+    if service_subsystem == subsystem then
```

Comment:
I'd prefer to invert this check and return a good error message.
That way we `return` for the happy case

Reply:
@james-callahan 

Do you consider logging it is enough? I mean that if we forcibly `return` the router rebuilding will never work. I know that the db should not be in that state ever, but still, I think it is better to just log it and move on.

Reply:
Added logging.

Reply:
logging is good. I'd still like to inverse this `if` statement.